py3status/modules/bluetooth.py

```diff
@@ -60,14 +61,23 @@ class Py3status:
     """
     # available configuration parameters
     cache_timeout = 10
-    device_separator = '|'
     format = DEFAULT_FORMAT
     format_device = '{name}'
+    format_separator = '\|'
 
     def post_config_hook(self):
-        # Do deprecation stuff here.  Because of the complexity we can't use
-        # the usual deprecation methods
-        # THIS IS A SPECIAL CASE DO NOT USE AS EXAMPLE CODE.
+        self.format_separator = self.py3.safe_format(self.format_separator)
+
+        # DEPRECATION WARNING. SPECIAL CASE. DO NOT USE THIS AS EXAMPLE.
+        old_separator = getattr(self, 'device_separator', None)
+        if old_separator:
+            old_separator = old_separator.replace('|', '\|')
+            self.format_separator = self.py3.safe_format(old_separator)
+            msg = 'DEPRECATION WARNING: you are using old style configuration '
+            msg += 'parameters you should update to use the new format.'
+            self.py3.log(msg)
```

Comment:
Yes, but to check for `|` in case somebody put that in their config. Not just replacing. We add backslash too.

And if we merge them together using `warn_the_user` flag so it only output once... This could stop the log. I pray nobody really use `device_separator` or they don't log things... If they do and they got annoyed, they'll update it. 👍 
```
        if self.format != DEFAULT_FORMAT:
            # The user has set a format using the new style format so we are
            # done here.
            return
```